src/main/java/edu/iu/terracotta/controller/app/AssessmentController.java

```diff
@@ -83,41 +84,44 @@ public ResponseEntity<List<AssessmentDto>> getAssessmentByTreatment(@PathVariabl
+    public ResponseEntity<AssessmentDto> getAssessment(@PathVariable long experimentId,
+                                                       @PathVariable long conditionId,
+                                                       @PathVariable long treatmentId,
+                                                       @PathVariable long assessmentId,
                                                        @RequestParam(name = "questions", defaultValue = "false") boolean questions,
                                                        @RequestParam(name = "answers", defaultValue = "false") boolean answers,
                                                        @RequestParam(name = "submissions", defaultValue = "false") boolean submissions,
                                                        @RequestParam(name = "submission_id", required = false) Long submissionId,
                                                        HttpServletRequest req)
             throws ExperimentNotMatchingException, BadTokenException, AssessmentNotMatchingException,
-            SubmissionNotMatchingException, NoSubmissionsException {
+                SubmissionNotMatchingException, NoSubmissionsException {
 
         SecuredInfo securedInfo = apijwtService.extractValues(req, false);
         apijwtService.experimentAllowed(securedInfo, experimentId);
         apijwtService.assessmentAllowed(securedInfo, experimentId, conditionId, treatmentId, assessmentId);
+
         if (submissionId != null) {
             apijwtService.submissionAllowed(securedInfo, assessmentId, submissionId);
         }
 
-        if(apijwtService.isLearnerOrHigher(securedInfo)) {
-            boolean student = !apijwtService.isInstructorOrHigher(securedInfo);
-            if (student && submissionId != null) {
-                // This will throw NoSubmissionsException if the submission doesn't belong to
-                // the student
-                this.submissionService.getSubmission(experimentId, securedInfo.getUserId(), submissionId, student);
-            }
-            AssessmentDto assessmentDto = assessmentService.toDto(assessmentService.getAssessment(assessmentId),
-                    submissionId, questions, answers, submissions, student);
-            return new ResponseEntity<>(assessmentDto, HttpStatus.OK);
-        } else {
+        if(!apijwtService.isLearnerOrHigher(securedInfo)) {
             return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
         }
+
+        boolean student = !apijwtService.isInstructorOrHigher(securedInfo);
+
+        if (student && submissionId != null) {
+            // This will throw NoSubmissionsException if the submission doesn't belong to
+            // the student
+            this.submissionService.getSubmission(experimentId, securedInfo.getUserId(), submissionId, student);
+        }
+
+        AssessmentDto assessmentDto = assessmentService.toDto(assessmentService.getAssessment(assessmentId),
```

Comment:
So, the old functionality of not exposing the correct answers when a student is accessing the API is pretty important. The REST API should never return which answers are the correct ones when the user is a student, with, at the moment, just a single exception. QuestionSubmissionController.getQuestionSubmissionsBySubmission does return correct answers for students, but it also checks assessment settings first to make sure that it is okay to do so (QuestionSubmissionServiceImpl.getQuestionSubmissions).

Reply:
I removed the student variable because it wasn't ACTUALLY being used in the service. In the AssessmentServiceImpl.toDto method, student was passed in, but never called.

The actual fix was line 198 in AssessmentServiceImpl.toDto... the rest is just clean-up of code.

Are you suggesting we add a new query param to the endpoint? I'm a bit confused... :)

Reply:
Or should Ryan hit the QuestionSubmissions controller instead of the Assessments controller for the data, as that does the check?

Reply:
> I removed the student variable because it wasn't ACTUALLY being used in the service. In the AssessmentServiceImpl.toDto method, student was passed in, but never called.

Ah, okay. Yeah, that's a regression. It should use student and not show answers if a student. I missed that in reviewing https://github.com/terracotta-education/terracotta/pull/330/files#diff-8e024878ff830441ad4b0681d702979f8952adba0e009003d6b2b8daf7b182ed

Can you please restore the use of the student parameter?